### ISSUE TITLE
WinAppDriver crashes when I call element.Text

### ISSUE THREAD

**Issue by jaska45** [Microsoft/WinAppDriver#405]
I use element.FindElementsByXPath("//*") to get all child elements of a message box dialog. Then when I go through the element list calling subElement.Text on each causes WinAppDriver to crash on 4th sub element. I can see that the correct GET .../text is called but that kills WinAppDriver .exe. My application is a Windows Forms app that calls MessageBox.Show to show a message box dialog.

My test code is

```
    [TestMethod]
    public void MessageBox()
    {
      var capabilities = new DesiredCapabilities();
      capabilities.SetCapability("app", @"D:\MessageBoxSample\bin\Debug\MessageBoxSample.exe");

      var driver = new WindowsDriver<WindowsElement>(new Uri("http://127.0.0.1:4723"), capabilities);

      var window = driver.FindElementByAccessibilityId("Form1");
      var button = window.FindElementByAccessibilityId("button1");
      button.Click();

      var messageBox = window.FindElementByClassName("#32770");
      var elements = messageBox.FindElementsByXPath("//*");

      foreach (var element in elements)
      {
        var str = element.Text;
      }
    }
```

You can find the MessageBoxSample.exe here
[MessageBoxSample.zip](https://github.com/Microsoft/WinAppDriver/files/1939905/MessageBoxSample.zip)

**Comment by zannett** [comment]
I am experiencing the same issue.

**Comment by timotiusmargo** [comment]
Hi @jaska45,

This is interesting. Would it be possible for you to get and share the log from the `WinAppDriver.exe` command console? For example:

```
==========================================
POST /session/E09441C3-9447-4003-8D98-BDAE053C11A4/elements HTTP/1.1
Accept: application/json, image/png
Content-Length: 36
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723

{"using":"xpath","value":"//Button"}
HTTP/1.1 200 OK
Content-Length: 280
Content-Type: application/json

{"sessionId":"E09441C3-9447-4003-8D98-BDAE053C11A4","status":0,"value":[{"ELEMENT":"42.5245984.4.1"},{"ELEMENT":"42.5245984.4.2"},{"ELEMENT":"42.5245984.4.3"},{"ELEMENT":"42.4918480.4.15"},{"ELEMENT":"42.4918480.4.19"},{"ELEMENT":"42.4918480.4.20"},{"ELEMENT":"42.4918480.4.21"}]}

==========================================
GET /session/E09441C3-9447-4003-8D98-BDAE053C11A4/element/42.5245984.4.1/text HTTP/1.1
Accept: application/json, image/png
Host: 127.0.0.1:4723


HTTP/1.1 200 OK
Content-Length: 97
Content-Type: application/json

{"sessionId":"E09441C3-9447-4003-8D98-BDAE053C11A4","status":0,"value":"Minimize Alarms & Clock"}
```

**Comment by jaska45** [comment]
```
Windows Application Driver listening for requests at: http://127.0.0.1:4723/
Press ENTER to exit.


==========================================
POST /session HTTP/1.1

Accept: application/json, image/png

Connection: Keep-Alive

Content-Length: 186

Content-Type: application/json;charset=utf-8

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)




HTTP/1.1 200 OK

Content-Length: 175

Content-Type: application/json



{"sessionId":"30BE4BFE-51BC-414A-AA56-592E5B6378D3","status":0,"value":{"app":"C:\\NewTool\\Bug\\MessageBoxSample\\bin\\Debug\\MessageBoxSample.exe","platformName":"Windows"}}


==========================================
POST /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element HTTP/1.1

Accept: application/json, image/png

Content-Length: 44

Content-Type: application/json;charset=utf-8

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)



{"using":"accessibility id","value":"Form1"}
HTTP/1.1 200 OK

Content-Length: 96

Content-Type: application/json



{"sessionId":"30BE4BFE-51BC-414A-AA56-592E5B6378D3","status":0,"value":{"ELEMENT":"42.1903462"}}


==========================================
POST /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element/42.1903462/element HTTP/1.1

Accept: application/json, image/png

Content-Length: 46

Content-Type: application/json;charset=utf-8

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)



{"using":"accessibility id","value":"button1"}
HTTP/1.1 200 OK

Content-Length: 96

Content-Type: application/json



{"sessionId":"30BE4BFE-51BC-414A-AA56-592E5B6378D3","status":0,"value":{"ELEMENT":"42.1247294"}}


==========================================
POST /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element/42.1247294/click HTTP/1.1

Accept: application/json, image/png

Content-Length: 2

Content-Type: application/json;charset=utf-8

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)



{}
HTTP/1.1 200 OK

Content-Length: 63

Content-Type: application/json



{"sessionId":"30BE4BFE-51BC-414A-AA56-592E5B6378D3","status":0}


==========================================
POST /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element/42.1903462/element HTTP/1.1

Accept: application/json, image/png

Content-Length: 39

Content-Type: application/json;charset=utf-8

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)



{"using":"class name","value":"#32770"}
HTTP/1.1 200 OK

Content-Length: 96

Content-Type: application/json



{"sessionId":"30BE4BFE-51BC-414A-AA56-592E5B6378D3","status":0,"value":{"ELEMENT":"42.2099436"}}


==========================================
POST /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element/42.2099436/elements HTTP/1.1

Accept: application/json, image/png

Content-Length: 31

Content-Type: application/json;charset=utf-8

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)



{"using":"xpath","value":"//*"}
HTTP/1.1 200 OK

Content-Length: 252

Content-Type: application/json



{"sessionId":"30BE4BFE-51BC-414A-AA56-592E5B6378D3","status":0,"value":[{"ELEMENT":"42.2099436"},{"ELEMENT":"42.1967926"},{"ELEMENT":"42.4131384"},{"ELEMENT":"42.2099436.2.-2147483647.2099436.-2.0"},{"ELEMENT":"42.2099436.2.-2147483647.2099436.-2.5"}]}


==========================================
GET /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element/42.2099436/text HTTP/1.1

Accept: application/json, image/png

Cache-Control: no-cache

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)




HTTP/1.1 200 OK

Content-Length: 74

Content-Type: application/json



{"sessionId":"30BE4BFE-51BC-414A-AA56-592E5B6378D3","status":0,"value":""}


==========================================
GET /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element/42.1967926/text HTTP/1.1

Accept: application/json, image/png

Cache-Control: no-cache

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)




HTTP/1.1 200 OK

Content-Length: 76

Content-Type: application/json



{"sessionId":"30BE4BFE-51BC-414A-AA56-592E5B6378D3","status":0,"value":"OK"}


==========================================
GET /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element/42.4131384/text HTTP/1.1

Accept: application/json, image/png

Cache-Control: no-cache

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)




HTTP/1.1 200 OK

Content-Length: 90

Content-Type: application/json



{"sessionId":"30BE4BFE-51BC-414A-AA56-592E5B6378D3","status":0,"value":"This is a sample"}


==========================================
GET /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element/42.2099436.2.-2147483647.2099436.-2.0/text HTTP/1.1

Accept: application/json, image/png

Cache-Control: no-cache

Host: 127.0.0.1:4723

User-Agent: selenium/3.11.0 (.net windows)
```

**Comment by jaska45** [comment]
This looks a bit odd

`GET /session/30BE4BFE-51BC-414A-AA56-592E5B6378D3/element/42.2099436.2.-2147483647.2099436.-2.0/text HTTP/1.1
`

The element id is different format than other.

**Comment by hassanuz** [comment]
Thanks @jaska45. We'll have to investigate this further.

**Comment by timotiusmargo** [comment]
Hi @jaska45,

Thanks for the reply. We are able to reproduce the issue and it is indeed a bug that we will fix in the next release. The **TitleBar** of the dialog window in the sample application you included above has no **Name** attribute (null). We will update our code to return empty string instead in such scenario.

For the time being, would you be able to update your application to properly populate the UIA name attribute of the dialog **TitleBar** above? Alternatively, you can slightly modify your test script to avoid calling the **Text** API endpoint above until we fix it in the next release. We'd be more than happy to suggest an alternative workaround if you can tell us what you are trying to do with the text you retrieved from the elements.

**Comment by jaska45** [comment]
Thanks. I can wait for the next version. The purpose of calling Text attribute of each element is just the collect the strings that are present on the UI at that moment. I am writing a localization testing tool. Previously be used UI automation on Windows but now studing if we can switch to WebDriver because we need use it anyway with web and mobile applications.

I am happy that you fix this because it is quite common that application just call Message.Show  with one parameter like in my sample.

```
    private void button1_Click(object sender, EventArgs e)
    {
      MessageBox.Show("This is a sample");
    }
```

I tested a case when I also pass Caption and it indeed works.

**Comment by timotiusmargo** [comment]
Hi @jaska45,

Thanks again for bringing this up. This issue was fixed in [v1.1-Preview](https://github.com/Microsoft/WinAppDriver/releases/tag/v1.1-Preview) release and newer. Please give it a try on the newest [releases](https://github.com/Microsoft/WinAppDriver/releases).

**Comment by Purushotham230** [comment]
Hi @timotiusmargo,  
i would like to get a text from an app from the username field after sending keys to my project login username field, Send keys are working, but I'm unable to get a text from it. I'm used getText(), and also .getAttribute("Value.Value") , but both are not working. do you know any other way to get a text? Otherwise, is there any way to verify username fields without getting a text from it and make assertions??
